### PR TITLE
JEC, b-tag SF and pileup for UL2018 samples

### PR DESCRIPTION
**Pull request by PascalBaertschi**
This commit adds JEC, b-tag SF and pileup files for Ultralegacy 2018 samples. The JEC files are taken from https://twiki.cern.ch/twiki/bin/view/CMS/JECDataMC#2018_Data and the b-tag SF files from https://twiki.cern.ch/twiki/bin/view/CMS/BtagRecommendation106XUL18.
The mc pileup distribution is created using information from Paul Lujan: https://github.com/cms-sw/cmssw/blob/ba6e8604a35283e39e89bc031766843d0afc3240/SimGeneral/MixingModule/python/mix_2018_25ns_UltraLegacy_PoissonOOTPU_cfi.py
The data pileup distribution is derived with the latest central pileup files and using the golden JSON of the UL2018 samples.

### CODE REVIEW

**Comment by gouskos** [comment]
Hi @PascalBaertschi the modifications on the code are good but there are some comments:
- BTV [@smoortga, @xcoubez]
1. Can you please include the UL2017 for DeepJet
2. It would be great if you could include the DeepCSV UL SF for 2017 and 2018 in this PR
- JME [@lathomas, @kirschen]: Could you please sign-off that the UL18_v5 are OK to be merged?

File: python/postprocessing/modules/btv/btagSFProducer.py [hunk 1]
```diff
@@ -130,6 +130,15 @@ def __init__(
                         2: "incl"   # light
                     },
                     'supported_wp': ["L", "M", "T", "shape_corr"]
+                },
```
Comment (gouskos) [inline python/postprocessing/modules/btv/btagSFProducer.py:133]: If possible you could add the DeepCSV UL SF around this line:
https://github.com/cms-nanoAOD/nanoAOD-tools/blob/ff0b5d7ceccc98b5f1317ffaa07e535813a2ccae/python/postprocessing/modules/btv/btagSFProducer.py#L104

Comment (PascalBaertschi) [reply python/postprocessing/modules/btv/btagSFProducer.py:160]: Hi @gouskos, I added the DeepCSV and DeepJet b-tag SF for UL2017 and the DeepCSV b-tag SF for UL2018 and changed btagSFProducer.py accordingly.

**Comment by lathomas** [comment]
The names of the files for UL18 JECs are ok (but I didn't explicitly check that they are what they claim to be). In fact I don't think it's very good practice to have duplicate files here. Those should be picked from the official JEC repository... 

I also realize that the JECs picked for UL2017 are not the latest ones (we have a Summer19UL17_v6 with updated residuals for AK8. AK4 are identical). @PascalBaertschi  Can you update that as well? 


Thanks !

**Comment by PascalBaertschi** [comment]
> The names of the files for UL18 JECs are ok (but I didn't explicitly check that they are what they claim to be). In fact I don't think it's very good practice to have duplicate files here. Those should be picked from the official JEC repository...
> 
> I also realize that the JECs picked for UL2017 are not the latest ones (we have a Summer19UL17_v6 with updated residuals for AK8. AK4 are identical). @PascalBaertschi Can you update that as well?
> 
> Thanks !

@lathomas Do you have a link to those files? Based on the list here: https://twiki.cern.ch/twiki/bin/view/CMS/JECDataMC it seems to me that version 5 is the latest, unless of course the list is not up to date.

**Comment by lathomas** [comment]
You can find them here: 
https://github.com/cms-jet/JECDatabase/tree/master/tarballs
The tag is Summer19UL17_X_V6. We will update the twiki, also with the global tag !

**Comment by PascalBaertschi** [comment]
Thanks, I updated the JEC UL2017 to version 6.

**Pull request merged**